pactman/verifier/verify.py

```
@@ -27,14 +31,27 @@ def __init__(self, pact, interaction, result_factory):
         self.response = ResponseVerifier(pact, interaction['response'], self.result)
 
     def __repr__(self):
-        return f"{self.pact.consumer}:{self.description}"
+        return f"<Interaction {self.pact.consumer}:{self.description}>"
+
+    def __str__(self):
+        return f"{self.pact.consumer} with request '{self.description}'"
 
     def verify(self, service_url, setup_url):
         self.result.start(self)
         try:
-            self.result.success = self.setup(setup_url)
-            if not self.result.success:
-                return False
+            self.set_provider_state_with_url(setup_url)
+            if self.result.success:
+                self.result.success = self.run_service(service_url)
+        finally:
+            self.result.end()
+
+    def verify_with_callable_setup(self, service_url, provider_setup):
+        self.result.start(self)
+        try:
+            try:
+                self.set_provider_state(provider_setup)
+            except Exception as e:
```

Should this catch something more specific like `ProviderStateMissing`?

---

IS it ok to silently swallow all the exceptions ?

---

Yeah, good point. Any other exception can be propagated up normally.